src/autoscaler/metricscollector/collector/app_streamer.go

```diff
@@ -66,19 +78,70 @@ func (as *appStreamer) streamMetrics() {
 
 		case event := <-eventChan:
 			as.processEvent(event)
+
+		case <-as.ticker.C():
+			as.computeAndSaveMetrics()
 		}
 	}
 }
 
 func (as *appStreamer) processEvent(event *events.Envelope) {
 	if event.GetEventType() == events.Envelope_ContainerMetric {
-		metric := models.GetInstanceMemoryMetricFromContainerMetricEvent(as.sclock.Now().UnixNano(), as.appId, event)
+		metric := noaa.GetInstanceMemoryMetricFromContainerMetricEvent(as.sclock.Now().UnixNano(), as.appId, event)
 		as.logger.Debug("process-event-get-memory-metric", lager.Data{"metric": metric})
 		if metric != nil {
 			err := as.database.SaveMetric(metric)
 			if err != nil {
 				as.logger.Error("process-event-save-metric", err, lager.Data{"metric": metric})
 			}
 		}
+	} else if event.GetEventType() == events.Envelope_HttpStartStop {
+		ss := event.GetHttpStartStop()
+		if ss != nil {
+			as.numRequests[ss.GetInstanceIndex()]++
+			as.sumReponseTimes[ss.GetInstanceIndex()] += (ss.GetStopTimestamp() - ss.GetStartTimestamp())
+		}
+	}
+}
+
+func (as *appStreamer) computeAndSaveMetrics() {
+	for instanceIdx, numReq := range as.numRequests {
+		if numReq != 0 {
+			througput := &models.AppInstanceMetric{
+				AppId:         as.appId,
+				InstanceIndex: uint32(instanceIdx),
+				CollectedAt:   as.sclock.Now().UnixNano(),
+				Name:          models.MetricNameThroughput,
+				Unit:          models.UnitRPS,
+				Value:         fmt.Sprintf("%.1f", float64(numReq)/as.collectInterval.Seconds()),
```

Comment:
should as.collectInterval be validated? As if it is zero, it will be panic here.